### PR TITLE
feat: Added DraftPaymentRequested filter.

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**
Implemented frontend-app-payment.filters.DraftPaymentRequested → titan.pipeline.GetTitanPayment

**JIRA:** [THES-181](https://2u-internal.atlassian.net/browse/THES-181)

### CODE REVIEW

**Review by grmartin — APPROVED** [review]

File: commerce_coordinator/apps/frontend_app_payment/filters.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ class PaymentRequested(OpenEdxPublicFilter):
     Filter to gather payment data from the defined PipelineStep(s)
     """
     # See pipeline step configuration OPEN_EDX_FILTERS_CONFIG dict in `settings/base.py`
-    filter_type = "org.edx.coordinator.frontend_app_ecommerce.payment.get.requested.v1"
+    filter_type = "org.edx.coordinator.frontend_app_payment.payment.get.requested.v1"
```
Comment (HammadAhmadWaqas) [inline commerce_coordinator/apps/frontend_app_payment/filters.py:21]: Updated this existing filter according to our spec.  I missed it before

File: commerce_coordinator/apps/frontend_app_payment/tests/test_filters.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        mock_get_payment.return_value = mock_get_payment_data
+
+        filter_params = {
+            'edx_lms_user_id': 1,
+        }
+
+        payment_details = DraftPaymentRequested.run_filter(filter_params)
+
+        expected_payment_details = dict(filter_params, **mock_get_payment_data)
+        self.assertEqual(payment_details, expected_payment_details)
```
Comment (HammadAhmadWaqas) [inline commerce_coordinator/apps/frontend_app_payment/tests/test_filters.py:31]: @pshiu Is this unittest enough to test what we have mentioned in the ticket?

Comment (pshiu) [reply commerce_coordinator/apps/frontend_app_payment/tests/test_filters.py:31]: Yes! Can you also add an `@override_settings` of `OPEN_EDX_FILTERS_CONFIG`?

**Review by pshiu — APPROVED** [review]

File: commerce_coordinator/apps/frontend_app_payment/tests/test_filters.py (hunk 2 above)
Comment (HammadAhmadWaqas) [reply commerce_coordinator/apps/frontend_app_payment/tests/test_filters.py:31]: @pshiu I am curious why do want to override settings.
To make sure the test doesn't fail when in the future we will update  OPEN_EDX_FILTERS_CONFIG in the base.py setting?
If that is the purpose, Should not we let the test fail if settings change? and update the test accordingly?

Comment (HammadAhmadWaqas) [reply commerce_coordinator/apps/frontend_app_payment/tests/test_filters.py:31]: hmmm. Make sense. Thank you for explaining.

File: commerce_coordinator/apps/frontend_app_payment/tests/test_filters.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    @override_settings(
+        OPEN_EDX_FILTERS_CONFIG={
+            "org.edx.coordinator.frontend_app_payment.payment.draft.requested.v1": {
+                "fail_silently": False,
+                "pipeline": [
+                    'commerce_coordinator.apps.titan.pipeline.GetTitanPayment',
+                ]
+            },
+        },
+    )
```
Comment (HammadAhmadWaqas) [inline commerce_coordinator/apps/frontend_app_payment/tests/test_filters.py:23]: @pshiu were we talking about this override?

Comment (pshiu) [reply commerce_coordinator/apps/frontend_app_payment/tests/test_filters.py:23]: Correct - thank you!

**Pull request merged**